src/h5web/visualizations/shared/AxisSystem.tsx

```diff
@@ -4,49 +4,49 @@ import { AxisLeft, AxisBottom } from '@vx/axis';
 import { format } from 'd3-format';
 import { TickRendererProps } from '@vx/axis/lib/types';
 import { Grid } from '@vx/grid';
-import { scaleLinear } from 'd3-scale';
 import styles from './AxisSystem.module.css';
 import { AxisOffsets, AxisDomains } from './models';
-import { adaptedNumTicks, sceneToAxisScales } from './utils';
+import { adaptedNumTicks, axisToDimsScale } from './utils';
 
 interface Props {
   axisDomains: AxisDomains;
   axisOffsets: AxisOffsets;
   showGrid?: boolean;
+  hasYLogScale?: boolean;
 }
 
 function AxisSystem(props: Props): JSX.Element {
-  const { axisDomains, axisOffsets, showGrid } = props;
+  const { axisDomains, axisOffsets, showGrid, hasYLogScale } = props;
 
   const { camera, size } = useThree();
   const { position, zoom } = camera;
   const { width, height } = size;
 
-  const [domains, setDomains] = useState<AxisDomains>({
+  // axisDomains are the complete domains. visibleDomains change with the camera
+  const [visibleDomains, setVisibleDomains] = useState<AxisDomains>({
     left: axisDomains.left,
     bottom: axisDomains.bottom,
   });
 
-  const sceneToAxis = sceneToAxisScales(size, axisDomains);
-
-  const axisScales = {
-    x: scaleLinear()
-      .domain(domains.bottom)
-      .range([0, width]),
-    y: scaleLinear()
-      .domain(domains.left)
-      .range([height, 0]),
+  const axisToCanvas = {
+    x: axisToDimsScale(axisDomains.bottom, [-width / 2, width / 2]),
+    y: axisToDimsScale(
```

Comment:
oh just realised that we use the term `dims` for the dimensions of the dataset -- perhaps `axisToSize` or `axisToScene`?